docs/_component/editor.client.js

```diff
@@ -198,6 +211,46 @@ export function Editor({children}) {
                 <code>remark-math</code>
               </a>
             </label>
+            <label>
+              <input
+                checked={state.directive}
+                type="checkbox"
+                onChange={() =>
+                  setConfig({...state, directive: !state.directive})
+                }
+              />{' '}
+              Use{' '}
+              <a href="https://github.com/remarkjs/remark-directive">
+                <code>remark-directive</code>
+              </a>
+            </label>
+            <hr />
+            <label>
+              <input
+                checked={state.formatMd}
+                type="checkbox"
+                onChange={() =>
+                  setConfig({...state, formatMd: !state.formatMd})
+                }
+              />{' '}
+              Use{' '}
+              <a href="https://mdxjs.com/packages/mdx/#optionsformat">
+                <code>format: &apos;md&apos;</code>
+              </a>
+            </label>
```

Comment:
Maybe we should add a tooltip here to explain MDX infers this based on the input file extension by default. I think this option makes sense for the playground, but it’s best to discourage using this option.

Actually, maybe a radio button is more appropriate here than a checkbox, although the checkbox currently fits nice in the UI.

Reply:
yes a radio or select can also work, open to change a bit the UI, not sure how to organize now that it's not just a list of remark plugins.

> but it’s best to discourage using this option.

Why? Docusaurus uses MDX and I think supporting a commonmark mode can help existing docs sites to adopt it more easily, ie to have something less strict running asap without a ton of compilation errors to fix

Reply:
> yes a radio or select can also work, open to change a bit the UI, not sure how to organize now that it's not just a list of remark plugins.

Let’s wait for @wooorm’s feedback, as I think he designed the website. I think it looks great already. :)

> > but it’s best to discourage using this option.
> 
> Why? Docusaurus uses MDX and I think supporting a commonmark mode can help existing docs sites to adopt it more easily, ie to have something less strict running asap without a ton of compilation errors to fix

This option changes whether files are treated as commonmark or MDX. Leaving it off, means this is inferred from the file extension. It’s not about strictness (although basically anything is technically valid markdown, but MDX has syntax errors). This option is useful when the file extension is unknown (such as in this playground), but I wouldn’t recommend it otherwise. I don’t know how Docusaurus uses it exactly though.

Reply:
On my side, I find it helpful to understand the MDX compiler behavior when handling .md files. 

I'm not really saying the `format` option is useful or recommended, and would be fine if instead we had the ability to customize the file name for example, as long as I can easily get the AST for an input parsed with CommonMark compat.

---

Although it's not recommended, Docusaurus historically has many sites that use MDX with a `.md` extension. One of the first unexpected things people stumble upon while upgrading to MDX 2 is that some tags (images, videos... ) do not appear anymore. See feedback here: https://github.com/facebook/docusaurus/discussions/8469#discussioncomment-5804253

For this reason, it's likely I'll add an option `format: 'mdx'`, to allow our users to upgrade their MDX version without having to rename all their files at the same time.

Reply:
You can pass the filename. [`compile()`](https://mdxjs.com/packages/mdx/#compilefile-options) takes a VFile compatible. In fact, I recommend always passing a filename if possible. Some plugins may depend on it. For example the JSX dev runtime depends on it.

```js
await compile({
  value: '# Hello <strong>world</strong>!',
  path: '/home/me/hello.mdx'
})
```

I would also be cautious treating `.md` files as MDX. MDX treats certain constructs differently.

I.e. given the following content:

```markdown
<div>
  foo

  bar
</div>
```

In markdown, this is equivalent to the following HTML:

```html
<div> foo bar </div>
```

In MDX, this is equivalen to the following JSX:

```jsx
<div>
  <p>foo</p>
  <p>bar</p>
</div>
```

I understand it’s useful to add the option for migration purposes in Docusaurus.

---

> it’s best to discourage using this option.

I think this may have been misinterpreted. I means it’s best to discourage using this option in general, but there are definitely legitimate use cases. The playground, where there are no actual file names, is a good use case for such an option.